utilities/storage.py

```
@@ -1146,6 +1146,22 @@ def verify_boot_sources_reimported(
         return False
 
 
+def wait_for_data_import_cron_data_sources_ready(admin_client: DynamicClient, namespace: str) -> None:
+    """Waits for all DataImportCron-managed DataSources to reach Ready=True.
+
+    Args:
+        admin_client: Cluster admin dynamic client.
+        namespace: Golden-images namespace name.
+    """
+    for data_source in get_data_sources_managed_by_data_import_cron(client=admin_client, namespace=namespace):
+        LOGGER.info(f"Waiting for DataSource {data_source.name} to be Ready")
```

Ruff (0.15.12)

[warning] Logging statement uses f-string

(G004)